src/ohri-form.component.tsx

```diff
@@ -202,14 +202,28 @@ const OHRIForm: React.FC<OHRIFormProps> = ({
             await Promise.all(
               postSubmissionHandlers.map(async ({ postAction, config, actionId }) => {
                 try {
-                  await postAction.applyAction(
-                    {
-                      patient,
-                      sessionMode,
-                      encounters: results.map((encounterResult) => encounterResult.data),
-                    },
-                    config,
-                  );
+                  const encounterData = [];
+                  if (results) {
+                    results.forEach((result) => {
+                      if (result?.data) {
+                        encounterData.push(result.data);
+                      }
+                    });
+                    if (encounterData.length) {
+                      await postAction.applyAction(
+                        {
+                          patient,
+                          sessionMode,
+                          encounters: results.map((encounterResult) => encounterResult.data),
```

Comment:
@kajambiya this is a duplication of code, `encounterData` already holds this information and you can use it here.

Reply:
nice catch. Was actually supposed to pass `encounterData` as the value for encounters